code/modules/mod/modules/module_kinesis.dm

```diff
@@ -0,0 +1,251 @@
+	kinesis_icon.appearance_flags = RESET_ALPHA|RESET_COLOR|RESET_TRANSFORM
+	grabbed_atom.add_overlay(kinesis_icon)
+	movement_animation = grabbed_atom.animate_movement
+	grabbed_atom.animate_movement = NO_STEPS
+	kinesis_beam = mod.wearer.Beam(grabbed_atom, "kinesis")
+	kinesis_catcher = mod.wearer.overlay_fullscreen("kinesis", /atom/movable/screen/fullscreen/kinesis, 0)
+	kinesis_catcher.kinesis_user = mod.wearer
+	kinesis_catcher.RegisterSignal(mod.wearer, COMSIG_MOVABLE_MOVED, /atom/movable/screen/fullscreen/kinesis.proc/on_move)
+	soundloop.start()
+
+/obj/item/mod/module/anomaly_locked/kinesis/on_deactivation()
+	. = ..()
+	if(!.)
+		return
+	clear_grab()
+
+/obj/item/mod/module/anomaly_locked/kinesis/process(delta_time)
+	if(!mod.wearer.client || mod.wearer.incapacitated(ignore_grab = TRUE))
+		clear_grab()
+		return
+	if(!range_check(grabbed_atom))
+		balloon_alert(mod.wearer, "out of range!")
+		clear_grab()
+		return
+	if(!kinesis_catcher.given_turf)
+		return
+	drain_power(use_power_cost/10)
+	mod.wearer.setDir(get_dir(mod.wearer, grabbed_atom))
+	if(grabbed_atom.loc == kinesis_catcher.given_turf)
+		if(grabbed_atom.pixel_x == kinesis_catcher.given_x - world.icon_size/2 && grabbed_atom.pixel_y == kinesis_catcher.given_y - world.icon_size/2)
+			return //spare us redrawing if we are standing still
+		animate(grabbed_atom, 0.2 SECONDS, pixel_x = grabbed_atom.base_pixel_x + kinesis_catcher.given_x - world.icon_size/2, pixel_y = grabbed_atom.base_pixel_y + kinesis_catcher.given_y - world.icon_size/2)
+		kinesis_beam.redrawing()
+		return
+	grabbed_atom.pixel_x = kinesis_catcher.given_x - world.icon_size/2
+	grabbed_atom.pixel_y = kinesis_catcher.given_y - world.icon_size/2
+	kinesis_beam.redrawing()
+	var/turf/next_turf = get_step_towards(grabbed_atom, kinesis_catcher.given_turf)
+	if(grabbed_atom.Move(next_turf))
+		if(isitem(grabbed_atom) && (mod.wearer in next_turf))
+			var/obj/item/grabbed_item = grabbed_atom
+			grabbed_item.pickup(mod.wearer)
+			mod.wearer.put_in_hands(grabbed_item)
+		return
+	var/direction = get_dir(grabbed_atom, next_turf)
+	if(direction & NORTH)
+		grabbed_atom.pixel_y = world.icon_size/2
+	else if(direction & SOUTH)
+		grabbed_atom.pixel_y = -world.icon_size/2
+	if(direction & EAST)
+		grabbed_atom.pixel_x = world.icon_size/2
+	else if(direction & WEST)
+		grabbed_atom.pixel_x = -world.icon_size/2
+	kinesis_beam.redrawing()
+	if(!isitem(grabbed_atom) || !COOLDOWN_FINISHED(src, hit_cooldown))
+		return
+	var/atom/hitting_atom
+	if(next_turf.density)
+		hitting_atom = next_turf
+	for(var/atom/movable/movable_content as anything in next_turf.contents)
+		if(ismob(movable_content))
+			continue
+		if(movable_content.density)
+			hitting_atom = movable_content
+			break
+	var/obj/item/grabbed_item = grabbed_atom
+	grabbed_item.melee_attack_chain(mod.wearer, hitting_atom)
+	COOLDOWN_START(src, hit_cooldown, hit_cooldown_time)
```

Comment:
There's a movement loop that does homing, I wonder if you wouldn't be better off using that. Would need to use signals to modify it slightly, but it'd work just fine, and I think would smooth out the snapping movement a bit. would also allow for a higher refresh rate then the process delay.

Making pixel_x/y work would be a bit more tricky, but not too bad

Reply:
Ahhh s harder then I thought. This is ok as a bare minimum